### ISSUE TITLE
How to show a preview when typing or moving cursor

### ISSUE THREAD

**Issue by andylokandy** [nushell/reedline#838]
Is it possible to print a preview (maybe multiline) when user is typing or moving the cursor, like a menu that the user can not interact with.

## Example

When editing, print the evaluation preview of the expression under the cursor:

```
~/Code/reedline〉(1+1█) *3
2
```

Once enter is hit, clear the preview, and the main program prints the result:

```
~/Code/reedline〉(1+1) *3
6
~/Code/reedline〉█
```

**Comment by fdncred** [comment]
sounds a bit dangerous. However, in `explore` there is `:try`

**Comment by andylokandy** [comment]
Yeah... But my use case is a simply math evaluator and in that case it is fine.

**Comment by sholderbach** [comment]
With the current design of reedline there are no provisions for that (parallel evaluation, with managed output).

From our side this has a low priority (we as Nushell would probably only use such facilities for other status output, without clear existing requirements for that yet).
So no guarantees that an addition in that direction will receive maintenance support.

Currently reedline's add-on components are all evaluated in line and in a blocking fashion. So if anything takes too long the UI slows down/freezes. We could accept something in the same vain without much redesign, but if you want that particular feature non-blocking we clearly have to prioritize the design of asynchronous updates for the more important components of reedline.
